<launch>
  <!-- Debug mode will start "troublesome" nodes in separate GDB instances. xterm is required. -->
  <arg name="debug_mode" value="0"/>

  <param name="/use_sim_time" value="false"/>

  <group ns="flower_mapper">
    <node pkg="manipulation_mapping"
          name="flower_mapper"
          type="flower_mapper_node"
          output="screen">

      <!-- Initial pose -->
      <rosparam param="init">
      </rosparam>

      <!-- Rates -->
      <rosparam file="$(find manipulation_mapping)/config/parameters.yaml"/>

    </node>
  </group>

  <!-- Debug: Nodes start in GDB in xterm -->
  <group if="$(eval arg('debug_mode')==1)">
    <node pkg="manipulation_vision"
          name="segmentation_node"
          type="segmentation_node"
          output="screen"
	  launch-prefix="xterm -e gdb --args">
    </node>

    <node pkg="manipulation_vision"
          name="classification_node"
          type="classification_node"
          output="screen"
          launch-prefix="xterm -e gdb --args">
    </node>

    <node pkg="manipulation_vision"
          name="search_node"
          type="search_node"
          output="screen"
          launch-prefix="xterm -e gdb --args">
    </node>
  </group>

  <!-- Non-debug mode -->
  <group if="$(eval arg('debug_mode')==0)">
    <node pkg="manipulation_vision"
          name="segmentation_node"
          type="segmentation_node"
          output="screen">
    </node>

    <node pkg="manipulation_vision"
          name="classification_node"
          type="classification_node"
          output="screen">
    </node>

    <node pkg="manipulation_vision"
          name="search_node"
          type="search_node"
          output="screen">
    </node>
  </group>
  
</launch>
